### ISSUE TITLE
`onSelected`: Keep default behavior of events

### ISSUE THREAD

**Issue by farzadab** [plotly/react-plotly.js#180]
Is there any way to keep the default behavior of events while providing a handler?

From what I understand from [this post](https://plotly.com/javascript/plotlyjs-events/#legend-click-events), unless the overridden function returns `false` the default behavior should be kept. However, when I provide an `onSelected` method the default behavior doesn't happen.
TBH, I'm not sure whether what I'm asking for is the expected behavior, but depending on the answer you can interpret this question as a feature request or a request for document clarification.

To make this more clear, what I see when `onSelected` is not provided is that the selected area stays active even after mouse is released and you can even select multiple regions like so:
![image](https://user-images.githubusercontent.com/5062458/80634084-b8091680-8a0e-11ea-933a-a4daac82ba02.png)
However, when `onSelected` is provided, the selected region disappears after the mouse is released and I cannot reproduce the same behavior (at least not so easily).
To give you more context, my specific use-case here is to be able to download the information of the selected points using a separate button that I have created, so the default visual response works well for me.

**Comment by giva9712** [comment]
@farzadab  Have you done? I am faced with the same problem. :(

**Comment by farzadab** [comment]
No, I didn't have many clues to go on really.

**Comment by bgriffen** [comment]
I also have the same issue, I have a  custom component in Retool ...

```javascript
onSelected={(v) => { 
                    modelUpdate({ selectedPoints: v.points.map(p => p.text)  })
        }}
onSelected={(v) => { 
                    modelUpdate({ selectedPoints: v.points.map(p => ({ x: p.x, y: p.y, well: p.text }) })
        }}
```

I want to simply be able to access the points highlighted in my model, which works(!) but the points not highlighted are no longer dimmed out so the user can't tell which wells are highlighted.

<img width="616" alt="Screen Shot 2022-02-28 at 2 44 54 pm" src="https://user-images.githubusercontent.com/2667869/156185253-9a35e495-74bc-42f4-b427-def9aea79834.png">

^ That is what I want to achieve but instead after a lasso highlight, all I get is the following v 

<img width="563" alt="Screen Shot 2022-02-28 at 2 44 23 pm" src="https://user-images.githubusercontent.com/2667869/156185356-4c23dfa4-71b3-415a-afeb-3fb25d634545.png">

It's like setting `onSelected` and `onClicked` just overwrites whatever default behaviour existed?

**Comment by RAJNEESH1708** [comment]
Have you resolved this issue? @farzadab @bgriffen

**Comment by akhan-calico** [comment]
Can we please resolve this ? Thanks

**Comment by bgriffen** [comment]
No, I didn't find any resolution. Unless it was swept up in another update elsewhere.

**Comment by akhan-calico** [comment]
Work around I am using feel free if anyone else has a better suggestion. 
Setup a ref value to track the state of the selected data etc.

I have a setup a useEffect which will run whenever selectedValue changes. 

Inside the useEffect I have a condition where it checks to see if the length of the selectedValues is greater than 0. If so then we set the ref.current to that if not we don't do anything. Then you can use that ref other places. 
You would update the setSelectedPoints to the callback which handles the event 

```  
const [selectedPoints, setSelectedPoints] = useState<DataPoint[]>([]);
  const pointsRef = useRef<DataPoint[]>([]);

  useEffect(() => {
    if (selectedPoints.length > 0) {
      pointsRef.current = selectedPoints;
    }
  }, [selectedPoints]);```
